lib/ufe/UsdAttribute.cpp

```diff
@@ -79,12 +80,20 @@ bool setUsdAttr(const PXR_NS::UsdAttribute& attr, const T& value)
 	return attr.Set<T>(value);
 }
 
-std::string getUsdAttributeValueAsString(const PXR_NS::UsdAttribute& attr)
+PXR_NS::UsdTimeCode getCurrentTime(const Ufe::SceneItem::Ptr& item)
+{
+	// Attributes with time samples will fail when calling Get with default time code.
+	// So we'll always use the current time when calling Get. If there are no time
+	// samples, it will fall-back to the default time code.
+	return MayaUsd::ufe::getTime(item->path());
```

Comment:
The way I approached this was to query the time via a virtual call to MayaUsdProxyShapeBase. The reason is that in the AL proxy shape, we compute the time value via the following equation:
```
outTime = timeOffset + timeScalar * time;
```
This was primarily to allow artists a bit more control over cache playback.

Reply:
FYI: these are the attributes that define our time value:
https://github.com/Autodesk/maya-usd/blob/dev/plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.h#L283

and this is where we compute the time: 
https://github.com/Autodesk/maya-usd/blob/dev/plugin/al/lib/AL_USDMaya/AL/usdmaya/nodes/ProxyShape.cpp#L1476

Reply:
@robthebloke please check the method we are having in utilities https://github.com/Autodesk/maya-usd/blob/dev/lib/ufe/Utils.cpp#L264. This method will call your version of getTime and time will be computed based on the equation you have mentioned.